ui/appui-react/src/appui-react/statusfields/SnapMode.tsx

```diff
@@ -149,8 +182,13 @@ export function SnapModeField(props: SnapModeFieldProps) {
                   props.onChange(item.value);
                   return;
                 }
-                // eslint-disable-next-line deprecation/deprecation
-                UiFramework.setAccudrawSnapMode(item.value);
+                // Bitwise XOR. Toggles the value in the bitmask. Allows multiple snap modes to be active at the same time.
+                const snapModeMask = snapMode ^ item.value;
```

Comment:
We can't change the bavior like that. Please document `onChange` to clarify, that the value in the callback is a single enum flag toggled on or off. Consumers can do the bit magic on their own.

Reply:
I just changed the default behaviour of the component, not the way onChange worked for API consumers. 

`props.onChange` is still called with only the new snap mode that has been toggled on. 
It fit's the param name `newSnapMode`.
 `onChange?: (newSnapMode: SnapMode) => void;`

Reply:
Oh, I see it now. However, behavior change is a breaking change as well i.e. current consumers might not want a multi-select behavior. Ideally, in such cases, an additional opt-in prop is added, however, I'd recommend that you use the `onChange` handler in your application and handle the bit-wise operations there since Redux state/dispatchers are deprecated.

Reply:
I supposed that supporting multi snap shoul've been the standard from the beginning. Since snapmode is a bitmask made with that purpose in mind. I dislike the idea of having the same component behaving differently in the apps. Especially components used for editing, like snaps. 
I guess this in more of a UX question. What should be the standard behaviour ?

I suggest we do the opposite, let consumers use the `onChange` handler if they dont want to support multi snap.

Reply:
The logic outside of `onChange` is considered deprecated and will be removed in 6.0 either way.
There's no way of knowing if any of the consumers depend on snap modes being single-select.

To be safe and to follow semantic versioning rules I'd suggest one of:
- Add an optional `multiple` flag to the component (will change both onChange and redux dispatch) - might not make sense long term if we want to support multi-select only (in the major version we could change the default value)
- Add a preview feature `multipleSnapModes` (once again, we can change the default in the next major version AND drop the preview feature)